core/modules/css/KvInterfaceImplMem.cc

```diff
@@ -92,14 +94,26 @@ KvInterfaceImplMem::KvInterfaceImplMem(string const& filename) {
 KvInterfaceImplMem::~KvInterfaceImplMem() {
 }
 
-void
-KvInterfaceImplMem::create(string const& key, string const& value) {
-    LOGF_DEBUG("create(%1%, %2%)" % key % value);
-    if (exists(key)) {
-        throw KeyExistsError(key);
+std::string
+KvInterfaceImplMem::create(string const& key, string const& value, bool unique) {
+    LOGF_DEBUG("create(%1%, %2%, unique=%3%)" % key % value % int(unique));
+
+    string path = key;
+    if (unique) {
+        // append unique suffix, in-memory KVI is not meant for large-scale
+        // storage, so we can do dumb brute-force loop
+        int seq = 0;
```

Comment:
That's what I was getting at. If seq is a random number between 1 and a million, there will pretty much never be a collision, and if there is, increment should get a safe value quickly. It's more complicated below.